sys/net/application_layer/dns/dns.c

```diff
@@ -157,21 +157,21 @@ static int _parse_dns_reply(uint8_t *buf, size_t len, void* addr_out, int family
             return -EBADMSG;
         }
         bufpos += RR_RDLENGTH_LENGTH;
-        if ((bufpos + addrlen) >= buflim) {
+        if ((bufpos + addrlen) > buflim) {
```

Comment:
This introduces an off-by-one/out-of-bound error. `buflim` is the first byte _after_ `buf` (= `buf + len`, see above), so if (`bufpos - buf + addrlen` is just one byte longer than `len`, you cause a buffer overflow with this.

Reply:
Huh that's odd. I would see the function always return error here, upon closer inspection I would get

```
2020-02-23 17:48:15,409 #  dns request riot-os.org
2020-02-23 17:48:15,416 # got 57 bytes from DNS
2020-02-23 17:48:15,418 # 0x3fff3efd + 16 < 0x3fff3f0d ?
```
or 
```
2020-02-23 17:48:07,632 #  dns request golem.de
2020-02-23 17:48:07,645 # got 54 bytes from DNS
2020-02-23 17:48:07,648 # 0x3fff3efa + 16 < 0x3fff3f0a ?
```

And I would get a valid address when reading the additional byte.

I'm giving that a closer look.

Reply:
Let's assume `len = addrlen` and `bufpos = buf` -> `buflim = buf + addrlen`

That means `memcpy(out, buf, len)` should be as valid as `memcpy(out, bufpos, addrlen)`, so the change is correct.

Reply:
Right, because `bufpos + addrlen` is also the first byte after the address.... Sorry for the noise...